### PR TITLE
147 delete all ast.Node validations

### PR DESCRIPTION
**Pull request by janishorsts**

### CODE REVIEW

File: parse/lex.go [hunk 1]
```diff
@@ -555,6 +555,14 @@ func lexIdentifier(l *lexer) stateFn {
 
 		switch {
 		default:
+			if len(s) == 0 && typ != itemMarkupClose {
+				return l.emitErrorf("missing %s name", typ)
+			}
+
+			if strings.HasSuffix(s, ":") {
+				return l.emitErrorf(`invalid identifier "%s"`, s)
```
Comment (janishorsts) [inline parse/lex.go:563]:
```suggestion
				return l.emitErrorf(`invalid %s name "%s"`, typ, s)
```

File: parse/parse.go [hunk 2]
```diff
@@ -397,7 +392,7 @@ func (p *parser) parseExpression() (Expression, error) {
 	case itemFunction, itemPrivateStart, itemReservedStart:
 		p.backup()
 	case itemExpressionClose: // empty expression
-		return expr, nil
+		return errorf("%w: missing operand or annotation", mf2.ErrSyntax)
```
Comment (janishorsts) [inline parse/parse.go:395]: Duplicate ErrSyntax.

https://github.com/expect-digital/go-mf2/pull/149/files#diff-5ed87adddc43e6af620e98ca4d4a40fe884b5d4b6bf1c9d7f2352f68f34328c4R156 already adds ErrSyntax.

```suggestion
		return errorf("missing operand or annotation")
```

File: parse/parse.go [hunk 3]
```diff
@@ -697,19 +692,36 @@ func (p *parser) parseMatcher() (Matcher, error) {
...
-	for itm := p.next(); itm.typ != itemEOF; itm = p.next() {
-		switch itm.typ {
+	// parse one or more selectors
+
+done:
+	for {
+		switch itm := p.nextNonWS(); itm.typ {
+		default:
+			p.backup()
+			break done
```
Comment (janishorsts) [inline parse/parse.go:702]: Missing case for EOF

```suggestion
			break done
		case itemEOF:
			return errorf(unexpectedErr(itm))
```

File: parse/parse_test.go [hunk 4]
```diff
@@ -379,15 +379,15 @@ func TestParseComplexMessage(t *testing.T) {
...
-.input { $input2 ^|quot| @b=c}
-.input { $input3 ! hey hey @c=1 @d=2}
-.local $local1={1}
-.local $local2={|2| ^private @a @b=2}
-.local $local3 = { > reserved}
-.reserved1 {$reserved1}
-.reserved2 hey |quot| hey { |reserved| :func }
-.reserved3 |body| |body2| {$expr1} {|expr2|} { :expr3 } { $expr4 ^hey @beep @boop}
-{{Text}}`,
+		.input { $input2 ^|quot| @b=c}
```
Comment (janishorsts) [inline parse/parse_test.go:382]: Put indentation back as it was before :)

File: parse/parse.go (hunk 3 above)
Comment (mvilks) [reply parse/parse.go:702]: Should we create a test for it?

Comment (janishorsts) [reply parse/parse.go:702]: Up to you

File: parse/parse.go [hunk 5]
```diff
@@ -697,19 +692,38 @@ func (p *parser) parseMatcher() (Matcher, error) {
...
+	}
+
+	// parse one or more variants
+
+	for {
+		switch itm := p.nextNonWS(); itm.typ {
+		default:
+			err := unexpectedErr(itm, itemCatchAllKey, itemNumberLiteral, itemQuotedLiteral, itemUnquotedLiteral)
+			return errorf("%w", err)
+		case itemEOF:
```
Comment (janishorsts) [inline parse/parse.go:724]: Missing case itemError?

File: parse/parse_test.go [hunk 6]
```diff
@@ -539,8 +539,8 @@ func TestParseComplexMessage(t *testing.T) {
 		{
 			name: "simple matcher with newline variants",
 			input: `.match { $variable :number }
-1 {{Hello { $variable } world}}
-* {{Hello { $variable } worlds}}`,
+		1 {{Hello { $variable } world}}
```
Comment (janishorsts) [inline parse/parse_test.go:542]: Can you please restore indentation as it was before?

**Review by mvilks — APPROVED** [review]

**Pull request merged**